donkeycar/parts/keras.py

```diff
@@ -261,3 +261,68 @@ def custom_sequential():
                   loss_weights={'angle_out': 0.6, 'throttle_out': 0.4})
 
     return model
+
+
+class CustomWithSpeed(KerasPilot):
+    def __init__(self, model=None, *args, **kwargs):
+        super(CustomWithSpeed, self).__init__(*args, **kwargs)
+        if model:
+            self.model = model
+        else:
+            self.model = custom_with_speed()
+
+        model_json = self.model.to_json(indent=2)
+        with open("latest_model.json", "w") as json_file:
+            json_file.write(model_json)
+            print('Saved model to JSON')
+
+    def inputs(self):
+        return ['cam/image_array', 'optic/speed']
+
+    def run(self, img_arr):
```

Comment:
This should also take the optic/speed as input and feed to neural network (requires some weird reshape)